scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,260 @@ output:
+
+Prior to integration all 10 samples went through the same filtering, normalization, and dimensionality reduction.
+These 10 samples were then merged into one `SingleCellExperiment` object following the same steps outlined in `03-dataset_integration.Rmd`.
+The merged object was then integrated with `fastMNN` to obtain a corrected gene expression assay and corrected reduced dimensionality results.
+The final SCE object was stored in `data/rms/integrated/rms_all_sce.rds`.
+
+We also have provided a metadata file, `data/rms/annotations/rms_sample_metadata.tsv`, that contains information from each sample, such as diagnosis, sex, age, etc.
+In this file each row corresponds to a sample found in the integrated SCE object.
+
+To begin, let's set up our directories and files:
+
+```{r filepaths}
 # set up file paths 
 # data directory for RMS data
 data_dir <- file.path("data", "rms")
 
-# integrated file containing samples to for DE 
+# integrated file containing all samples to use for DE analysis
 integrated_sce_file <- file.path(data_dir, "integrated", "rms_all_sce.rds")
 
-# sample metadata with information needed to setup DE
+# sample metadata with information needed to setup DE analysis
 sample_metadata_file <- file.path(data_dir, "annotations", "rms_sample_metadata.tsv")
 
-# output files 
+# directory to store output
 deseq_dir <- file.path("analysis", "rms", "deseq")
 if(!dir.exists(deseq_dir)){
   dir.create(deseq_dir, recursive = TRUE)
 }
 
-# results from deseq
+# results file to output from DE analysis
 deseq_output_file <- file.path(deseq_dir, "rms_myoblast_deseq_results.tsv")
 ```
 
+We can go ahead and read in the SCE object and the metadata file.
 
-```{r}
-# read in integrated SCE object 
+```{r read files}
+# read in the SCE object that has already been integrated
 integrated_sce <- readr::read_rds(integrated_sce_file)
 
 # read in sample metadata file 
 sample_metadata <- readr::read_tsv(sample_metadata_file)
 ```
 
-Before we set up our experiment, let's explore our dataset a little.
+## Dataset Exploration
 
-```{r}
-# First just plot by sample
-scater::plotReducedDim(integrated_sce,
-                       dimred = "fastmnn_UMAP",
-                       colour_by = "sample",
-                       point_size= 0.5,
-                       point_alpha = 0.2)
+Before we dive into differential expression, let's explore our integrated SCE object and the dataset a little more.
+
+We'll start by looking at what's inside the object.
+Here we should have both the original (uncorrected) data and the integrated (corrected) data for both the gene expression and the reduced dimensionality results.
+How are those stored in our object?
+
+```{r print sce}
+# print out entire object
+integrated_sce
+
+# look at the assay names in our object
+assayNames(integrated_sce)
+
+# look at the names of the dimensionality reduction present in the SCE object
+reducedDimNames(integrated_sce)
 ```
```

Comment:
These shoudl probably be separate chunks, to keep the output separate when running interactively. When rendered, the chunk will be broken by the outputs, but in an interactive session, all three will appear in the same place.